packages/neptune-extension/src/labextension/index.tsx

```diff
@@ -29,7 +31,14 @@ class Extension implements DocumentRegistry.IWidgetExtension<NotebookPanel, INot
 
     context.ready.then(() => {
       const platformNotebook = new Notebook(context, this.app);
-      ReactDOM.render(<App platformNotebook={platformNotebook} />, widget.node);
+
+      const store = configureStore();
+
+      ReactDOM.render((
+        <Provider store={store}>
```

Comment:
Please move the store logic to a separate component like e.g. `Bootstrap`. The code does not belong to extension initialization and is duplicated across extensions.

Reply:
```
ReactDOM.render(<Bootstrap platformNotebook={platformNotebook} />, ...)
------
Bootstrap = (props) => (
  <Provider>
    <App {...props} />
  </Provider>
)
```

Reply:
Agree.